contracts/ssnlist.scilla

```diff
@@ -326,73 +344,56 @@ end
 procedure FillInDepositDeleg(ssnaddr: ByStr20, deleg : ByStr20)
   FillInDepositDelegAmt ssnaddr deleg _amount
 end
-procedure FillInDelegsWithAmt(ssnaddr : ByStr20, deleg : ByStr20, deleg_amt : Uint128)
-  d <- delegs[ssnaddr][deleg];
-  match d with
-  | Some amt =>
-    new_amt_deleg = builtin add deleg_amt amt;
-    delegs[ssnaddr][deleg] := new_amt_deleg
-  | None =>
-    delegs[ssnaddr][deleg] := deleg_amt
-  end
-end
-procedure FillInDelegs(ssnaddr : ByStr20, deleg : ByStr20)
-  amt = _amount;
-  FillInDelegsWithAmt ssnaddr deleg _amount
-end
 procedure HasRewardToWithdraw(ssnaddr : ByStr20, deleg : ByStr20)
   lrcd_tmp <- last_withdraw_cycle_deleg[deleg][ssnaddr];
   lrcd = option_uint128_value lrcd_tmp;
   lrc <- lastrewardcycle;
   has_reward = builtin lt lrcd lrc;
   match has_reward with
   | True =>
-    throw
+    e = DelegHasUnwithdrawRewards;
+    ThrowError e
+  | False =>
+  end
+end
+procedure HasBufferedDeposit(ssnaddr : ByStr20, deleg : ByStr20)
+  ldcd_o <- last_buf_deposit_cycle_deleg[deleg][ssnaddr];
+  ldcd = option_uint128_value ldcd_o;
+  lrc <- lastrewardcycle;
+  (* in fact, lrc will only more than or equal to ldcd, equal means he delegates this around and withdraw this around as well *)
+  has_buffered = uint128_le lrc ldcd;
+  match has_buffered with
+  | True =>
+    e = DelegHasBufferedDeposit;
+    ThrowError e
   | False =>
   end
 end
 procedure DistributeStakeAmt(initiator: ByStr20, ssn: ByStr20)
```

Comment:
Change distribute to withdrawal. Distribute is usually refer to rewards distribution

Reply:
ok